lms/djangoapps/courseware/courses.py

```diff
@@ -94,9 +93,39 @@ def get_course_with_access(user, action, course_key, depth=0, check_if_enrolled=
     check_if_enrolled: If true, additionally verifies that the user is either enrolled in the course
       or has staff access.
     """
-    assert isinstance(course_key, CourseKey)
-    course = get_course_by_id(course_key, depth=depth)
-    access_response = has_access(user, action, course, course_key)
+    course = get_course_by_id(course_key, depth)
+    check_course_access(course, user, action, check_if_enrolled)
+    return course
+
+
+def get_course_overview_with_access(user, action, course_key, check_if_enrolled=False):
+    """
+    Given a course_key, look up the corresponding course overview,
+    check that the user has the access to perform the specified action
+    on the course, and return the course overview.
+
+    Raises a 404 if the course_key is invalid, or the user doesn't have access.
+
+    check_if_enrolled: If true, additionally verifies that the user is either enrolled in the course
+      or has staff access.
+    """
+    try:
+        course_overview = CourseOverview.get_from_id(course_key)
+    except CourseOverview.DoesNotExist:
+        raise Http404("Course not found.")
+    check_course_access(course_overview, user, action, check_if_enrolled)
+    return course_overview
+
+
+def check_course_access(course, user, action, check_if_enrolled=False):
```

Comment:
Again, I'm not a big fan of switching order of params unless there is a good reason.  I would instead go for consistency with has_access call below, unless there is a good reason to change both to begin with course.